src/System.Security.Cryptography.X509Certificates/tests/X500DistinguishedNameTests.cs

```diff
@@ -126,6 +126,38 @@ public static void NotQuotedWithQuotes(string quoted, string notQuoted, string h
 
             Assert.Equal(notQuoted, dn.Decode(X500DistinguishedNameFlags.DoNotUseQuotes));
         }
+        
+        [Theory]
+        [MemberData(nameof(T61Cases))]
+        public static void T61Strings(string expected, string hexEncoded)
+        {
+            byte[] encoded = hexEncoded.HexToByteArray();
+            X500DistinguishedName dn = new X500DistinguishedName(encoded);
+
+            Assert.Equal(expected, dn.Name);
+        }
+
+        [Theory]
+        [MemberData(nameof(T61CasesUtf8))]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        public static void T61StringsUtf8(string expected, string hexEncoded)
+        {
+            byte[] encoded = hexEncoded.HexToByteArray();
+            X500DistinguishedName dn = new X500DistinguishedName(encoded);
+
+            Assert.Equal(expected, dn.Name);
+        }
+
+        [Theory]
+        [MemberData(nameof(T61CasesLatin1))]
+        [PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.OSX)]
+        public static void T61StringsLatin1(string expected, string hexEncoded)
```

Comment:
I think I'm fine with dumping `X500NameEncoder.OpenSslDecode.cs` as part of this change, and just using the same managed implementation on macOS and Linux.  It'll cut down on a lot of SafeHandles and P/Invokes, as well as unify the behaviors.

Reply:
Will do.